cmd/prune-junit-xml/logparse/logparse_test.go

```diff
@@ -0,0 +1,123 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logparse
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestParse(t *testing.T) {
+	t.Logf("Full regular expression:\n%s", klogPrefix.String())
+
+	testcases := map[string]struct {
+		log           string
+		expectEntries []Entry
+	}{
+		"empty": {},
+
+		"one-other": {
+			log: "other",
+			expectEntries: []Entry{
+				&OtherEntry{Data: "other"},
+			},
+		},
+
+		"one-klog": {
+			log: `I1007 13:16:55.727802 1146763 example.go:57] "Key/value encoding" logger="example"`,
+			expectEntries: []Entry{
+				&KlogEntry{
+					Data:     `I1007 13:16:55.727802 1146763 example.go:57] "Key/value encoding" logger="example"`,
+					Severity: SeverityInfo,
+				},
+			},
+		},
+
+		"one-unit": {
+			log: `example_test.go:45: E1007 13:28:21.908998] hello world`,
+			expectEntries: []Entry{
+				&KlogEntry{
+					Data:     `example_test.go:45: E1007 13:28:21.908998] hello world`,
+					Severity: SeverityError,
+				},
+			},
+		},
+		"mixture": {
+			log: `other
+I1007 13:16:55.727802 1146763 example.go:57] "a"
+   E1007 13:16:55.727802 1146763 example.go:58] "b" foo=<
+   	indented
+    >
+middle
+example_test.go:45: E1007 13:28:21.908998] hello world
+`,
+			expectEntries: []Entry{
+				&OtherEntry{Data: "other\n"},
+				&KlogEntry{
+					Data: `I1007 13:16:55.727802 1146763 example.go:57] "a"
+`,
+					Severity: SeverityInfo,
+				},
+				&KlogEntry{
+					Data: `   E1007 13:16:55.727802 1146763 example.go:58] "b" foo=<
+   	indented
+    >
+`, Severity: SeverityError,
+				},
+				&OtherEntry{Data: "middle\n"},
+				&KlogEntry{
+					Data: `example_test.go:45: E1007 13:28:21.908998] hello world
+`,
+					Severity: SeverityError,
+				},
+			},
+		},
+	}
+
+	for name, tc := range testcases {
+		t.Run(name, func(t *testing.T) {
+			actualEntries := Parse(strings.NewReader(tc.log))
+			require.Equal(t, tc.expectEntries, actualEntries)
+
+			var buf strings.Builder
+			for _, entry := range actualEntries {
+				_, _ = buf.WriteString(entry.String())
+			}
+			require.Equal(t, tc.log, buf.String())
+
+			for i := 0; i < len(tc.expectEntries); i++ {
+				t.Run(fmt.Sprintf("stop-after-%d", i), func(t *testing.T) {
+					var actualEntries []Entry
+					seq := All(strings.NewReader(tc.log))
+					e := 0
+					seq(func(entry Entry) bool {
+						actualEntries = append(actualEntries, entry)
+						if e >= i {
+							return false
+						}
+						e++
+						return true
+					})
+					require.Equal(t, tc.expectEntries[0:i+1], actualEntries)
+				})
+			}
+		})
+	}
+}
```

Comment:
no have other place to comment, but you are commiting a binary :) 

> cmd/prune-junit-xml/prune-junit-xml

Reply:
I, or rather my script, noticed: https://github.com/kubernetes/kubernetes/pull/127929#issuecomment-2400353443

This came from "go run ./cmd/prune-junit-xml" + "git add ./cmd/prune-junit-xml".